api/v1alpha3/azurecluster_default.go

```diff
@@ -89,7 +89,7 @@ func (c *AzureCluster) setSubnetDefaults() {
 		cpSubnet.SecurityGroup.Name = generateControlPlaneSecurityGroupName(c.ObjectMeta.Name)
 	}
 	if cpSubnet.RouteTable.Name == "" {
-		cpSubnet.RouteTable.Name = generateRouteTableName(c.ObjectMeta.Name)
+		cpSubnet.RouteTable.Name = generateControlPlaneRouteTableName(c.ObjectMeta.Name)
```

Comment:
I think we should consider not creating a control plane route table by default and only creating it if the user specifies one since the default use case doesn't require one

Reply:
so still use node route table unless the user specifies their own name of control plane route table?

Reply:
yes, the route table is attached to the subnet and the cloud provider cares about the NodeSubnet so it should only care about the NodeRouteTable. This behavior is just more correct because it doesn't ignore the Spec for route table on the  control plane subnet and allows more advanced users to customize their cluster. But the default should stay the same, ie. Node subnet has a route table, control plane subnet doesn't, and the azure.json file has the node route table as routeTableName on all nodes.